File: Data Structures & Algorithms/depth-of-binary-tree/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        if(!root) return 0;
        let q = new Queue([root]);
        let level = 0;
        while(!q.isEmpty()){
            let size = q.size();

            for(let i=0; i<size;i++){
                 let node  = q.pop();
                 if(node.left)  q.push(node.left);
                 if(node.right)  q.push(node.right);
            }
            level++;
        }
        return level;
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        self.maxSum = float('-inf')

        def dfs(node) -> int:
            if not node:
                return 0
            
            leftMax = max(dfs(node.left),0)
            rightMax = max(dfs(node.right),0)

            curr_path_sum = node.val + leftMax + rightMax

            self.maxSum = max(curr_path_sum, self.maxSum)
            return node.val + max(leftMax, rightMax)
        dfs(root)
        return self.maxSum